File: 프로그래머스/lv2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
def isPrime(n):
    if n == 1:
        return False
    for i in range(2, int(n ** 0.5) + 1):
        if n % i == 0:
            return False
    return True


def convert(n, q):
    rev_base = ''

    while n > 0:
        n, mod = divmod(n, q)
        rev_base += str(mod)

    return rev_base[::-1]


def solution(n, k):
    cnt = 0
    k진법의수 = convert(n, k)
    nums = str(k진법의수).split('0')
    for num in nums:
        if num and isPrime(int(num)):
            cnt += 1

    return cnt

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.js
function isPrime(n) {
  if (n < 2) {
    return false;
  }
  for (let i = 2; i < parseInt(n ** 0.5) + 1; i++) {
    if (n % i === 0) {
      return false;
    }
  }
  return true;
}

function convert(n, base) {
  const ans = [];
  while (n > 0) {
    ans.push(n % base);
    n = parseInt(n / base);
  }
  return ans.reverse().join('');
}

function solution(n, k) {
  let cnt = 0;
  const k진법의수 = convert(n, k);
  const nums = k진법의수.split('0');
  for (const num of nums) {
    if (num && isPrime(parseInt(num))) {
      cnt += 1;
    }
  }

  return cnt;
}

